### PR TITLE
hypre: add support for building shared and static libs with external BLAS/LAPACK

### PR DESCRIPTION
**Pull request by lexming**
This update will support `configopts` in the form
* `configopts = ''`
* `configopts = '--enable-shared'`
* `configopts = ['', '--enable-shared'`]

### CODE REVIEW

File: easybuild/easyblocks/h/hypre.py [hunk 1]
```diff
@@ -37,25 +38,51 @@
 class EB_Hypre(ConfigureMake):
     """Support for building Hypre."""
 
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor."""
+
+        super(EB_Hypre, self).__init__(*args, **kwargs)
+
+        self.config_shared = False
+        self.config_static = False
```
Comment (Micket) [inline easybuild/easyblocks/h/hypre.py:47]: This flag is mutually exclusive anyway right? So, we can just keep one flag, shared(true) or not(false)? `self.enable_shared = False`

**Review by Micket — CHANGES_REQUESTED** [review]

File: easybuild/easyblocks/h/hypre.py (hunk 1 above)
Comment (lexming) [reply easybuild/easyblocks/h/hypre.py:47]: Nope, they can be both `True` if built with `configopts = ['', '--enable-shared']`. Then the sanity check is performed for both the static and shared libs.

**Comment by Micket** [comment]
I actually wrote a comment reply, but, apparently, i forgot to actually "send" in the review, so it was stuck as pending; my question was simply; "does easybuild only run the sanity check once when looping?"

I'll assume you are right on that one to not hold this back any longer.

**Comment by lexming** [comment]
@Micket yes, the sanity checks are run only once regardless of the number of iterations.

**Pull request merged**